cmd/cloud_sql_proxy/cloud_sql_proxy.go

```diff
@@ -501,16 +505,41 @@ func main() {
 	}
 	logging.Infof("Ready for new connections")
 
-	(&proxy.Client{
+	proxyClient := &proxy.Client{
 		Port:           port,
 		MaxConnections: *maxConnections,
 		Certs: certs.NewCertSourceOpts(client, certs.RemoteOpts{
-			APIBasePath:  *host,
-			IgnoreRegion: !*checkRegion,
-			UserAgent:    userAgentFromVersionString(),
-			IPAddrTypeOpts:   ipAddrTypeOptsInput,
+			APIBasePath:    *host,
+			IgnoreRegion:   !*checkRegion,
+			UserAgent:      userAgentFromVersionString(),
+			IPAddrTypeOpts: ipAddrTypeOptsInput,
 		}),
 		Conns:              connset,
 		RefreshCfgThrottle: refreshCfgThrottle,
-	}).Run(connSrc)
+	}
+
+	if *usegRPC {
+		logging.Infof("Using gRPC\n")
+
+		for conn := range connSrc {
+			rpcProxy, err := grpcproxy.ObtainProxyConnection(grpcproxy.AuthConfig{
+				ProxyClient: proxyClient,
+				Conn:        conn,
+				Port:        grpcPort})
+			if err != nil {
```

Comment:
throwing a fatal error if any connection fails seems pretty harsh. Consider just logging the error?